executor/oci/mounts.go

```diff
@@ -76,11 +76,11 @@ func withROBind(src, dest string) func(m []specs.Mount) ([]specs.Mount, error) {
 
 func hasPrefix(p, prefixDir string) bool {
 	prefixDir = filepath.Clean(prefixDir)
-	if prefixDir == "/" {
+	if filepath.Base(prefixDir) == string(filepath.Separator) {
```

Comment:
this doesn't look the same as before and also I think this is `false` for all inputs

Reply:
Why would it be false? `filepath.Base("/")` is specified to return `filepath.Separator`. It also returns `filepath.Separator` if you call `filepath.Base("////")`, but we already called `filepath.Clean` so that won't be the case.

The old code could never match on Windows, because `filepath.Clean` on Windows will never have `/` in it, and the simple change of `prefixDir == string(filepath.Separator)` fails because if `filepath.Clean` is given a path starting with `C:`, will retain the `C:` in the output.

Do you have a situation in-mind where the result is different? I had started implementing volume-path removal, and discovered that I was just open-coding `filepath.Base` anyway.

The other approach I considered is to say this function can only take `/`-style paths, and hence the fix is to use `path` instead of `filepath`. I think that's the case in current usage, because this is only used on the _destination_ of mounts, and currently only for Linux mounts. Making it `/`-only would mean that if this was used for Windows mounts, we'd have to pre-process those into drive-less `/` format, and possibly change them back when putting them into the OCI Spec.

Reply:
Ah sorry, I forgot that `filepath.Base("/")` returns `/`. I thought everything ending with `/` returns an empty string.